src/sync.cpp

```diff
@@ -511,10 +511,127 @@ bool assignFilesystemIds(Sync& sync, MegaApp& app, FileSystemAccess& fsaccess, h
+    if (!mTable)
+    {
+        assert(false);
+        LOG_err << "Unable to open DB table: " << dbname;
+        return;
+    }
+
+    mTable->rewind();
+
+    decltype(mNextTableId) tableId;
+    std::string data;
+    while (mTable->next(&tableId, &data))
+    {
+        auto syncConfig = SyncConfig::unserialize(data);
+        if (!syncConfig)
+        {
+            assert(false);
+            LOG_err << "Unable to unserialize sync config at id: " << tableId;
+            continue;
+        }
+
+        mSyncConfigs.insert(std::make_pair(syncConfig->getLocalPath(), SyncConfigData{tableId, *syncConfig}));
+        if (tableId > mNextTableId)
+        {
+            mNextTableId = tableId;
+        }
+    }
+    ++mNextTableId;
+}
+
+void SyncConfigBag::add(const SyncConfig& syncConfig)
+{
+    auto syncConfigPair = mSyncConfigs.find(syncConfig.getLocalPath());
+    if (syncConfigPair == mSyncConfigs.end())
+    {
+        mSyncConfigs.insert(std::make_pair(syncConfig.getLocalPath(), SyncConfigData{mNextTableId, syncConfig}));
+        if (mTable)
+        {
+            auto data = syncConfig.serialize();
+            if (!mTable->put(mNextTableId, &data))
+            {
+                assert(false);
+                LOG_err << "Incomplete database put at id: " << mNextTableId;
+            }
+            ++mNextTableId;
+        }
+    }
+    else
+    {
+        update(syncConfig);
+    }
+}
+
+void SyncConfigBag::remove(const SyncConfig& syncConfig)
+{
+    auto syncConfigPair = mSyncConfigs.find(syncConfig.getLocalPath());
+    if (syncConfigPair != mSyncConfigs.end())
+    {
+        if (mTable)
+        {
+            if (!mTable->del(syncConfigPair->second.mTableId))
+            {
+                assert(false);
+                LOG_err << "Incomplete database del at id: " << syncConfigPair->second.mTableId;
+            }
+        }
+        mSyncConfigs.erase(syncConfig.getLocalPath());
+    }
+    else
+    {
+        assert(false && "sync config does not exist");
+    }
+}
+
+void SyncConfigBag::update(const SyncConfig& syncConfig)
+{
+    auto syncConfigPair = mSyncConfigs.find(syncConfig.getLocalPath());
+    if (syncConfigPair != mSyncConfigs.end())
+    {
+        syncConfigPair->second.mSyncConfig = syncConfig;
+        if (mTable)
+        {
+            DBTableTransactionCommitter committer{mTable.get()};
+            const auto tableId = syncConfigPair->second.mTableId;
+            if (!mTable->del(tableId))
+            {
+                assert(false);
+                LOG_err << "Incomplete database del at id: " << tableId;
+            }
+            auto data = syncConfig.serialize();
+            if (!mTable->put(tableId, &data))
+            {
+                assert(false);
+                LOG_err << "Incomplete database put at id: " << tableId;
+            }
+        }
+    }
+    else
+    {
+        assert(false && "sync config does not exist");
```

Comment:
Perhaps log an error message is also a good idea for debugging purposes (based on logs, rather on a debugger :) )